### DISCUSSION TITLE
v2.00.0 Beta 1

### DISCUSSION THREAD

**Discussion by unexpectedpanda** [unexpectedpanda/retool#234] [Announcements]
**[Download Retool 2.00.0](https://unexpectedpanda.github.io/retool/download/)**

Retool 2.0 is here in beta form. Ten months in the making, it'll be in beta for
a few more months to get feedback and fix bugs found by users. Retool 1.x will
still be available during this time, although it won't receive clone list
updates from this point.

Retool 2.0 is a hefty rewrite with a focus on static typing, better code
practices, maintainability, and making things go _faster_.

View the [release notes](https://github.com/unexpectedpanda/retool/releases/tag/v2.00.0).

**Comment by unexpectedpanda** [comment]
Known issues:
- ~Running `--update` from the CLI the first time gets hijacked if `internal-config`/`datafile.dtd` are missing. Workaround: run it a second time to get the updates.~
- ~There's no prompt if you're missing clone lists or metadata the first time you run Retool, it relies on the user to update manually.~
- The **Process DAT files** button won't enable unless you have a region added in the global settings. This is by design and normally isn't an issue, but it means you can't only use system settings.
- ~No-Intro NES: _World Class Track Meet (USA) (Rev 1)_ and _Super Mario Bros. + Duck Hunt + World Class Track Meet (USA) (Rev 1)_ are both kept, as the compilations algorithm goes to a tie breaker when processing _World Class Track Meet_, which means the single title is selected. For the other titles in the compilation, the compilation is chosen.~
- ~The NES DAT is slow to process. The slowness is during compilations processing, and might be unavoidable due to some large (20-in-1, 15-in-1) compilations in the clone list that have crossovers.~

**Reply by zach-morris** [reply]
Is there an option for choosing/preferring compilations vs individual titles?  I wasn't able to figure it out and would prefer in most cases to keep the individual titles (Super Mario Bros, Duck Hunt, World Class Track Meet) and compilations seperately.  Edit:  I see the issue for the enhancement now.

**Reply by unexpectedpanda** [reply]
Yep, working on it at the moment. Dealing with compilations is the trickiest code in Retool, and I hate it with a passion 😂

When I'm done, preferring individual titles will be the default.

**Reply by unexpectedpanda** [reply]
Individual titles are now preferred over compilations, except where compilations have a higher primary region or primary language priority, or higher priority defined in a clone list.

**Comment by rufotheone-fr** [comment]
CONGRATULATIONS \o/
Can't wait to test it!

**Comment by FidoFuz** [comment]
I use retool commandline via a batch file.  v1 now won't update and v2 doesn't seem to support CL.

**Reply by unexpectedpanda** [reply]
[Retool v2 absolutely supports command line operation](https://unexpectedpanda.github.io/retool/how-to-use-retool-cli/). It's very similar to v1's operation.

If you want to continue running v1, make sure you have the [latest version](https://unexpectedpanda.github.io/files/retool-1.18-win-x86-64.zip) so it downloads the necessary files from the correct repository, and not the differently formatted files for v2. Be aware that v1 is no longer receiving clone list updates.

**Reply by FidoFuz** [reply]
Maybe it's just my inexperience with Python but I can't get the source to execute
![image](https://user-images.githubusercontent.com/4922434/226881026-eadfcf1e-3fe9-482e-8307-88178c4d13fd.png)

**Reply by unexpectedpanda** [reply]
Hard to tell from the cropped image, but it looks like it's having issues with type hint generics. This likely means you're running an old version of Python. Retool requires 3.10 or higher.

It's also possible you have multiple versions of Python installed and calling the py file directly is referencing an older installation.

**Reply by FidoFuz** [reply]
I've got 3.8 installed which was ok for Retool v1.  Installing latest python now, thanks.
